app/controllers/application_controller.rb

```diff
@@ -36,8 +36,12 @@ def authenticated?
     current_user.present?
   end
 
+  def save_redirect_path
+    session[:redirect_back_to] = request.original_fullpath
+  end
+
   def authenticate
-    redirect_to sign_in_path unless authenticated?
+    (save_redirect_path && (redirect_to sign_in_path)) unless authenticated?
```

Comment:
Might be tidier to extract a method here. (or if you don't want to do that I don't think we need all the brackets).

```
save_requested_path_and_redirect unless authenticated?
```

and 

```
def save_requested_path_and_redirect
  save_requested_path
  redirect_to sign_in_path
end
```